ENGAGEHF/HealthSummary/HealthSummaryView.swift

```diff
@@ -14,38 +14,84 @@ import SwiftUI
 
 
 struct HealthSummaryView: View {
+    private enum ShareState {
+        case pdf
+        case qrCode
+    }
+    
     @State private var healthSummaryDocument: PDFDocument?
     @State private var viewState: ViewState = .idle
+    @State private var fullUrl: String?
+    @State private var code: String?
+    @State private var shareState: ShareState = .pdf
     
     
     var body: some View {
         NavigationStack {
-            ZStack {
-                if let healthSummaryDocument {
-                    PDFViewer(document: healthSummaryDocument)
+            VStack {
+                if shareState == .pdf {
+                    pdfView
                 } else {
-                    ProgressView("Generating Health Summary")
+                    qrCodeView
                 }
             }
-                .task {
-                    await self.generateHealthSummary()
-                }
                 .toolbar {
-                    if let healthSummaryDocument {
-                        ToolbarItem(placement: .confirmationAction) {
-                            ShareLink(
-                                item: healthSummaryDocument,
-                                preview: SharePreview("Health Summary", image: Image(.engagehfIcon))
-                            )
-                                .accessibilityLabel("Share Link")
-                        }
-                    }
+                    shareButton
+                    shareModeSelector
                 }
                 .viewStateAlert(state: $viewState)
         }
     }
     
     
+    private var pdfView: some View {
+        ZStack {
+            if let healthSummaryDocument {
+                PDFViewer(document: healthSummaryDocument)
+            } else {
+                ProgressView("Generating Health Summary")
+            }
+        }
+        .task {
+            await generateHealthSummary()
+        }
+    }
+    
+    private var qrCodeView: some View {
+        ZStack {
+            if let fullUrl, let code {
```

Comment:
If I understood it correctly, we won't be using the fullUrl anymore - please rename to `url` and only make use of that property from the response as well. You should even already receive it given the current implementation.